tests/bluetooth/ctrl_isoal/CMakeLists.txt

```diff
@@ -15,4 +15,6 @@ zephyr_library_include_directories(
 
 FILE(GLOB app_sources src/*.c)
 
+add_compile_definitions(CONFIG_BT_CTLR_ISOAL_SINKS=4 CONFIG_BT_CTLR_ISOAL_SOURCES=4)
```

Comment:
I believe this should go into prj.conf

Reply:
This is my suggestion. The problem with using `prj.conf` is that you may end up activating a lot more code than you want due to dependencies. A unit test needs to be able to have full control of its dependencies, and for that reason I suggested to allow this kind of override.

Reply:
Thank you for the explanation. I found some precedence in other CMakeLists.txt for test cases, so I will approve.

Reply:
It looks that this is another approach to get full control over configuration by tests. 
Here you can see another approach that was used for LLCPs: https://github.com/zephyrproject-rtos/zephyr/pull/40073#issuecomment-1044510294.

@tejlmand have you got any suggestion how to follow with this king of issues in general?

Reply:
> The problem with using `prj.conf` is that you may end up activating a lot more code than you want due to dependencies.

No, what you have to do in this case is to take the needed Kconfig settings into a local `Kconfig` file to make them locally available and thus removing all dependencies, like this:
```
# tests/bluetooth/ctrl_isoal/Kconfig

config BT_CTLR_ISOAL_SINKS
        int "Number of Isochronous Adaptation Layer sinks (for unittest)"

config BT_CTLR_ISOAL_SOURCES
        int "Number of Isochronous Adaptation Layer sinks (for unittest)"

source "Kconfig.zephyr"
```

This will then allow you to set the test values in the standard way using `prj.conf`, like this:
```
# prj.conf
CONFIG_BT_CTLR_ISOAL_SINKS=4
CONFIG_BT_CTLR_ISOAL_SOURCES=4
....
```

A little more work, but it keeps the Kconfig infrastructure intact and all settings correctly available to CMake, autoconf.h and any python script relying on those settings.
Also developers will know where to look for configuration settings.

Finally, your current approach only sets it as compile flag.
This might be enough in this particular case, but is by-passing the normal flow and creating a mess wrt. the design.

Reply:
Thanks. I did not know that the Kconfig could be locally overridden. That's a very nice solution.

Reply:
thanks for applying the proposal.